esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst

```diff
@@ -0,0 +1,258 @@
+            - light.turn_on:
+                id: green_led_light
+                transition_length: 0.5s
+            - delay: 0.5s
+            - light.turn_off:
+                id: green_led_light
+                transition_length: 0.5s
+            - delay: 0.5s
+            - light.turn_on:
+                id: green_led_light
+                transition_length: 0.5s
+            - delay: 0.5s
+            - light.turn_off:
+                id: green_led_light
+                transition_length: 0.5s
+            - delay: 0.5s
+            - light.turn_on:
+                id: green_led_light
+                transition_length: 0.5s
+            - delay: 0.5s
+            - light.turn_off:
+                id: green_led_light
+                transition_length: 0.5s
+            - lambda: |-  
+                id(triple_click).publish_state(false);
+
+        - timing: # Single Long Clicked
+            - ON for 1s to 3s
+            - OFF for at least 0.5s
+          then:
+            - logger.log: "Single Long Clicked"
+            - lambda: |-  
+                id(long_click).publish_state(true);
+            - light.turn_on:
+                id: green_led_light
+                transition_length: 0.5s
+            - delay: 0.5s
+            - light.turn_off:
+                id: green_led_light
+                transition_length: 0.5s
+            - lambda: |-  
+                id(long_click).publish_state(false);
+
+        - timing: # Single Short Clicked
+            - ON for at most 1s
+            - OFF for at least 0.5s
+          then:
+            - logger.log: "Single Short Clicked"
+            - switch.toggle: relais
+
+
+      - platform: status
+        name: "$devicename Status"
+
+      - platform: template
+        name: "$devicename Double Click"
+        id: double_click
+        lambda: |-
+          return false;    
+
+      - platform: template
+        name: "$devicename Triple Click"
+        id: triple_click
+        lambda: |-
+          return false;  
+
+      - platform: template
+        name: "$devicename Long Click"
+        id: long_click
+        lambda: |-
+          return false;   
+
+
+    sensor:
+      - platform: wifi_signal
+        name: "$devicename WiFi Signal"
+        update_interval: 60s
+    #    expire_after: 0
+
+    switch:        #relais
+      - platform: gpio
+        name: "$devicename"
+        pin: GPIO12
+        id: relais
+
+    output:
+      # Register the green LED as a dimmable output ....
+      - platform: esp8266_pwm
+        id: green_led
+        pin:
+          number: GPIO13
+          inverted: True
+
+    light:
+      - platform: monochromatic
+        name: "Green LED"
+        output: green_led
+        id: green_led_light
+        internal: True
+
+
+
+3. Home Assistant
+*******************
+
+An example for an automation that toggles a switch when another switch is double clicked:
+
+.. code-block:: yaml
+    - alias: on double click
```

Comment:
Insert an empty line after `.. code-block:: yaml`. Otherwise it will not show correctly.